app/src/main/java/com/xmartlabs/template/controller/ServiceProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.xmartlabs.template.controller;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.xmartlabs.template.model.EntityWithId;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import retrofit2.Response;
+
+public interface ServiceProvider<T, D extends EntityWithId<T>> {
```

Comment:
I agree. I guess that in some case would be valid to have a service provider without an entity, in such case maybe you can define a similar interface without the entity stuff. What do you think?